### PR TITLE
Added locationResponseVO to vendor response info

### PR DESCRIPTION
**Pull request by Bozhokmaria**
Added locationResponseVO to vendor response info as frontend team asked

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java [hunk 1]
```diff
@@ -25,7 +26,7 @@ public VendorMapper(LocationService locationService) {
 
     public VendorResponseVO toVO(Vendor vendor) {
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java:27]: please add tests to check that mapping is correct

File: src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java [hunk 2]
```diff
@@ -47,15 +48,5 @@ private void provideLocationDependencies(VendorRequestVO request, Vendor vendor)
 
     private void configureModelMapper() {
         modelMapper.getConfiguration().setMatchingStrategy(MatchingStrategies.STRICT);
-        modelMapper.addMappings(createSkipPropertyMap());
-    }
-
-    private PropertyMap<VendorRequestVO, Vendor> createSkipPropertyMap() {
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java]: why it is removed?

File: src/main/java/com/exadel/discountwebapp/vendor/vo/VendorRequestVO.java [hunk 3]
```diff
@@ -3,9 +3,13 @@
...
 
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+
 @EqualsAndHashCode(callSuper = true)
 @Data
 @SuperBuilder
 @NoArgsConstructor
 public class VendorRequestVO extends VendorBaseVO {
+    private Long locationId;
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/vendor/vo/VendorRequestVO.java:15]: @NotNull? (no need to define @Positive)

File: src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java [hunk 4]
```diff
@@ -25,7 +26,7 @@ public VendorMapper(LocationService locationService) {
 
     public VendorResponseVO toVO(Vendor vendor) {
         VendorResponseVO response = modelMapper.map(vendor, VendorResponseVO.class);
-        response.setLocationId(vendor.getLocation().getId());
+        response.setLocationResponseVO(modelMapper.map(vendor.getLocation(), LocationResponseVO.class));
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java]: probably LocationMapper can be injected and reused here to map Location -> LocationResponseVO

File: src/main/java/com/exadel/discountwebapp/vendor/vo/VendorResponseVO.java [hunk 5]
```diff
@@ -10,4 +11,6 @@
 @AllArgsConstructor
 public class VendorResponseVO extends VendorBaseVO {
     private Long id;
-}
+
+    private LocationResponseVO locationResponseVO;
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/vendor/vo/VendorResponseVO.java]: let's name this field simply 'location'

Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/vendor/vo/VendorResponseVO.java]: +

File: src/main/java/com/exadel/discountwebapp/vendor/vo/VendorRequestVO.java (hunk 3 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/vendor/vo/VendorRequestVO.java:15]: +

File: src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java (hunk 4 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java]: +

File: src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java (hunk 1 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/vendor/mapper/VendorMapper.java:27]: added

**Pull request merged**